tests/capi/unittest_capi_service_db_mock.cc

```diff
@@ -0,0 +1,477 @@
+ * @param[in] options LevelDB options
+ * @param[in] name database path
+ * @param[out] errptr error message if error occurs
+ * @return leveldb_t* database handle
+ */
+leveldb_t* leveldb_open (const leveldb_options_t* options,
+    const char* name, char** errptr)
+{
+  return mockInstance->leveldb_open (options, name, errptr);
+}
+/**
+ * @brief Mocking function for leveldb_put() of the LevelDB
+ * @param[in] db database handle.
+ * @param[in] options write option.
+ * @param[in] key key string.
+ * @param[in] keylen length of key string.
+ * @param[in] val value string.
+ * @param[in] vallen length of value string.
+ * @param[out] errptr error message if error occurs.
+ */
+void leveldb_put (leveldb_t* db, const leveldb_writeoptions_t* options,
+    const char* key, size_t keylen, const char* val, size_t vallen, char** errptr)
+{
+  return mockInstance->leveldb_put (db, options, key, keylen, val, vallen, errptr);
+}
+
+/**
+ * @brief Mocking function for leveldb_get() of the LevelDB
+ * @param[in] db database handle.
+ * @param[in] options read option.
+ * @param[in] key key string.
+ * @param[in] keylen length of key string.
+ * @param[out] vallen length of value string.
+ * @param[out] errptr error message if error occurs.
+ * @return char* value string if found. Otherwise NULL.
+ */
+char* leveldb_get(leveldb_t* db, const leveldb_readoptions_t* options,
+    const char* key, size_t keylen, size_t* vallen, char** errptr)
+{
+  return mockInstance->leveldb_get (db, options, key, keylen, vallen, errptr);
+}
+
+/**
+ * @brief Mocking function for leveldb_delete() of the LevelDB
+ * @param[in] db database handle.
+ * @param[in] options write option.
+ * @param[in] key key string.
+ * @param[in] keylen length of key string.
+ * @param[out] errptr error message if error occurs.
+ */
+void leveldb_delete(leveldb_t* db, const leveldb_writeoptions_t* options,
+    const char* key, size_t keylen, char** errptr)
+{
+  mockInstance->leveldb_delete (db, options, key, keylen, errptr);
+}
+
+/**
+ * @brief Mocking function for leveldb_close() of the LevelDB
+ * @param db database handle.
+ */
+void leveldb_close (leveldb_t* db)
+{
+  mockInstance->leveldb_close (db);
+}
+
+/**
+ * @brief Check the normal process of setting the pipeline description with a given name.
+ */
+TEST (MLServiceAPIDBMockTest, set_pipeline_description_0_p)
+{
+  int ret = 0;
+  const gchar *key = "ServiceName";
+  gchar *pipeline = g_strdup_printf ("appsrc name=appsrc ! "
+      "other/tensor,dimension=(string)1:1:1:1,type=(string)float32,framerate=(fraction)0/1 ! "
+      "tensor_sink name=tensor_sink");
+  mockInstance = new LevelDBMock();
+
+  EXPECT_CALL (*mockInstance, leveldb_open(_, _, _))
+      .Times(1).WillOnce(Return((leveldb_t*)0x1234));
+  EXPECT_CALL (*mockInstance, leveldb_put(_, _, _, _, _, _, _))
+      .Times(1).WillOnce(SetArgPointee<6>(nullptr));
+  EXPECT_CALL (*mockInstance, leveldb_close(_))
+      .WillRepeatedly(Return());
```

Comment:
Can't we know how many times this function will be called?

Reply:
Updated!